gatherers/manifest.js

```diff
@@ -54,34 +38,87 @@ class Manifest extends Gather {
     };
   }
 
+  /**
+   * @private
+   */
+  static _createDeferred() {
+    let resolve;
+    let reject;
+    const promise = new Promise((internalResolve, internalReject) => {
```

Comment:
Seconded, we need to find an alternative path here. Not sure what that is just yet, but this feels very much like thin ice.